app/api/v2/handlers/operation_api.py

```diff
@@ -47,3 +73,24 @@ async def get_operation_report(self, request: web.Request):
         access = await self.get_request_permissions(request)
         report = await self._api_manager.get_operation_report(operation_id, access)
         return web.json_response(report)
+
+    '''Overridden Methods'''
+    async def create_object(self, request: web.Request):
+        data = await request.json()
+        await self._error_if_object_with_id_exists(data.get(self.id_property))
+        access = await self.get_request_permissions(request)
+        return await self._api_manager.create_object_from_schema(self.schema, data, access)
+
+    async def create_or_update_object(self, request: web.Request):
+        data, access, obj_id, query, search = await self._parse_common_data_from_request(request)
+        matched_obj = self._api_manager.find_object(self.ram_key, query)
+        if matched_obj and matched_obj.access not in access['access']:
+            raise JsonHttpForbidden(f'Cannot update {self.description} due to insufficient permissions: {obj_id}')
+        return await self._api_manager.create_object_from_schema(self.schema, data, access, matched_obj)
```

Comment:
should `matched_obj` just be a search dict with the id?

Reply:
`matched_obj` is an instance of the existing object. This method is called on both PUT/POST requests, with `matched_obj` being an optional parameter. It's used to check if we can replace an object on a PUT request - if the operation is already in a finished state, it won't allow us to replace it